features/fields.feature

```diff
@@ -24,10 +24,9 @@ Functionaliteit: Aanpasbare representatie met de fields parameter
 
   Attributen binnen een groep kunnen ook individueel worden bevraagd. De dot-notatie wordt gebruikt om specifieke attributen van een groep te selecteren. Bijvoorbeeld de voornamen, geboortedatum en geboorteplaats van een persoon kunnen worden opgevraagd via fields=naam.voornamen,geboorte.datum,geboorte.plaats.
 
-  Met de fields parameter kan ook worden aangegeven welke relaties moeten worden opgenomen. Dit betreft de links (in _links) die verwijzen (uri) naar de betreffende gerelateerde resource. Wanneer de fields parameter is meegegeven in het request worden alleen die relaties teruggegeven die zijn gevraagd in de fields parameter.
-  In de fields parameter kan de relatie worden gevraagd door de relatienaam op te nemen al dan niet voorafgegaan door HAL-element _links.
-  Bijvoorbeeld de links naar de kinderen van een persoon worden teruggegeven bij fields=burgerservicenummer,naam,kinderen. In dat geval worden andere relaties, zoals ouders en partners niet opgenomen in het antwoord (in _links). Hetzelfde antwoord kan worden verkregen met fields=burgerservicenummer,naam,_links.kinderen.
-  Wanneer de resource een property bevat met exact dezelfde naam als een van de properties in _links, dan wordt bij het vragen van deze property alleen het property uit de resource teruggegeven en niet het gelijknamige property uit _links. In dat geval kan de gelijknamige relatie alleen worden gevraagd door _links voor de linkpropertynaam te plaatsen.
+  Met de fields parameter kunnen velden uit een groep gevraagd worden zonder de groepsnaam op te nemen. Bijvoorbeeld fields=voornamen levert naam.voornamen. Bijvoorbeeld de links naar de kinderen van een persoon worden teruggegeven bij fields=burgerservicenummer,naam,kinderen. In dat geval worden andere relaties, zoals ouders en partners niet opgenomen in het antwoord (in _links). Hetzelfde antwoord kan worden verkregen met fields=burgerservicenummer,naam,_links.kinderen.
+  Wanneer een veld in een groep niet uniek is binnen de resource, en vragen van het veld met fields zonder de groepsnaam levert niet één uniek veld op, wordt een foutmelding gegeven.
+  Wanneer een veld voorkomt in de resource en in een groep in de resource komt een veld voor met dezelfde naam, dan wordt het veld met exact de opgegeven naam teruggegeven. Bijvoorbeeld als een resource een property woonadres heeft en ook een link _links.woonadres, dan levert fields=woonadres het woonadres, maar niet _links.woonadres.
```

Comment:
Ik blijf het toch een beetje een vreemde feature vinden, als ik het goed begrijp wil HC het volgende?
Als een property alleen voorkomt als property van de resource en niet als property van een sub groep (waaronder _links), dan geven we de property terug als er wordt gevraagd om de property naam (zonder pad).
Als een property voorkomt als property van de resource en als property van een sub groep dan geven we de property van de resource terug, als daarom wordt gevraagd (zonder pad) en/of
als er wordt gevraagd om property van sub groepen (met pad), dan leveren we die ook.
Als een property niet voorkomt als property van de resources, maar wel één keer voorkomt als property van een sub groep, dan geven we die terug als er om de property wordt gevraagd zonder pad.
Als een property niet voorkomt als property van de resource, maar wel als property van meerdere sub groepen, dan geven we een foutmelding als er wordt gevraagd om de property zonder pad, maar niet als je zoekt met de property naam met pad. Dat laatste staat er niet expliciet, maar daar ga ik vanuit.

Reply:
Volgens mij begrijp je het goed. 
Iemand kan een gevraagd veld opgeven met:
- het volledige pad naar dat veld
- het kortst mogelijke pad dat het veld uniek identificeert

Ik begrijp niet zo goed waarom dit vreemd is, dit is functionaliteit die XPath, jsonpath en css selectors ook kennen, met alleen dat verschil dat we niet toestaan dat er meer dan één veld gevonden wordt. Deze toevoeging is alleen gedaan omdat in BRK velden met dezelfde naam op heel veel plekken kunnen voorkomen, zoals code, omschrijving, identificatie, aantekening, stukIdentificatie. Daarom vereisen we dat de selectie in één uniek veld moet resulteren.
Deze functionaliteit voorkomt bovendien dat de waarde in de fields parameter enorm lang wordt.

Of bedoel je dat het vreemd is opgeschreven?

Ter illustratie van hoe het moet werken een voorbeeld resource huisdieren
```
{
  "soort": "hond",
  "ras": "tekkel",
  "naam": "Fikkie",
  "eigenaar": {
    "identificatie": {
      "nummer": 1524,
      "soort": "volgnummer"
    },
    "naam": {
      "voornaam": "Mark",
      "achternaam": "Strijker"
    }
  },
  "adres": {
    "straat": "Het laantje",
    "nummer": 3
  },
  "identificatie": "k1jh23k-123kj4h-1k23j5h4"
}
```
| fields | resultaat |
| --- | --- |
| ras | { "ras": "tekkel" } |
| naam | "naam": "Fikkie" |
| eigenaar.naam | { "eigenaar": { "naam": { "voornaam": "Mark", "achternaam": "Strijker" }  } } |
| naam.achternaam | { "eigenaar": { "naam": { "achternaam": "Strijker" }  } } |
| eigenaar.identificatie.nummer | { "eigenaar": { "identificatie": { "nummer": 1524 } } } |
| identificatie.nummer | { "eigenaar": { "identificatie": { "nummer": 1524 } } } |
| adres.nummer | { "adres": { "nummer": 3 } } |
| nummer | Foutmelding 400 (fields is niet uniek) |